src/env.rs

```diff
@@ -46,10 +47,14 @@ pub fn get_remaining_points_for_instance(instance: &Instance) -> u64 {
     }
 }
 
-pub fn sub_remaining_point(env: &Env, points: u64) -> anyhow::Result<()> {
+pub fn sub_remaining_point(env: &Env, points: u64) {
     let instance = &env.instance.clone().unwrap();
-    set_remaining_points(instance, get_remaining_points_for_env(env) - points);
-    Ok(())
+    let remaining_points = get_remaining_points_for_env(env);
+    if let Some(remaining_points) = remaining_points.checked_sub(points) {
+        set_remaining_points(instance, remaining_points);
+    } else {
+        abi_bail!("Remaining point reach zero");
```

Comment:
I think we should make the naming uniform. Point => gas. But that can be a followup